client/question/services/FeedbackGeneratorService.js

```diff
@@ -263,33 +328,70 @@ tie.factory('FeedbackGeneratorService', [
      * @param {string|null} outputFunctionName If needed, is string of the
      *    name for the function needed to process/format the user's output.
      * @param {CorrectnessTest} TestCase
+     * @param {number} testCaseIndex
      * @param {*} observedOutput Actual output for running user's code.
+     * @param {string} testSuiteId
      * @returns {Feedback}
      * @private
      */
     var _getCorrectnessTestFeedback = function(
-      outputFunctionName, testCase, observedOutput) {
+        outputFunctionName, testCase, testCaseIndex, observedOutput,
+        testSuiteId) {
+      var testCaseStateKey = testSuiteId + '-' + testCaseIndex;
       var allowedOutputExample = testCase.getAnyAllowedOutput();
       var feedback = FeedbackObjectFactory.create(
         FEEDBACK_CATEGORIES.INCORRECT_OUTPUT_FAILURE);
-      // TODO(eyurko): Add varied statements for when code is incorrect.
-      feedback.appendTextParagraph('Your code produced the following result:');
-      if (outputFunctionName) {
-        feedback.appendCodeParagraph(
-          'Input: ' + _jsToHumanReadable(testCase.getInput()) + '\n' +
-          ('Output (after running ' + outputFunctionName + '): ' +
-            _jsToHumanReadable(observedOutput)));
-        feedback.appendTextParagraph(
-          'However, the expected output of ' + outputFunctionName + ' is:');
+      if (testSuiteId == lastTestSuiteId) {
+        if (correctnessTestStates[testCaseStateKey] ==
+            CORRECTNESS_TEST_INPUT_DISPLAYED &&
+            testSuiteId != SAMPLE_INPUT) {
+          // Display expected output
+          feedback.appendTextParagraph(_getCorrectnessFeedbackString(
+            CORRECTNESS_FEEDBACK_TEXT,
+            'EXPECTED_OUTPUT'));
+          feedback.appendCodeParagraph(
+            'Input: ' + _jsToHumanReadable(testCase.getInput()) + '\n' +
+            'Expected Output: ' + _jsToHumanReadable(allowedOutputExample));
+          correctnessTestStates[testCaseStateKey] =
+            CORRECTNESS_TEST_EXPECTED_OUTPUT_DISPLAYED;
+          return feedback;
+        }
       } else {
-        feedback.appendCodeParagraph(
-          'Input: ' + _jsToHumanReadable(testCase.getInput()) + '\n' +
-          'Output: ' + _jsToHumanReadable(observedOutput));
-        feedback.appendTextParagraph('However, the expected output is:');
+        // New / next test suite
+        lastTestSuiteId = testSuiteId;
+        if (testSuiteId != SAMPLE_INPUT) {
+          if (!(testCaseStateKey in correctnessTestStates)) {
+            // Display input user should try
+            feedback.appendTextParagraph(
+              _getCorrectnessFeedbackString(CORRECTNESS_FEEDBACK_TEXT,
+              'INPUT_TO_TRY'));
+            feedback.appendCodeParagraph(
+              'Input: ' + _jsToHumanReadable(testCase.getInput()));
+            correctnessTestStates[testSuiteId] = 1;
```

Comment:
This looks incorrect...

Reply:
Forgot to delete that line. Thanks for the catch.